ppcs/ppc0025-perl-version.md

```diff
@@ -0,0 +1,165 @@
+
+3. Move to Perl&#160;7 based on some headline feature (such as a new object system)
+
+   This has the same technical costs as moving to Perl&#160;7 generally, but the outside-world optics of the move are less strange. However, the feature then needs to be big enough to justify a bump in this way, lest it be perceived as a PR stunt for repackaging mostly-same-old as a major new version. And it does not set a precedent for when the version is to be bumped, opening up a future debate about whether changes are big enough to warrant a bump, and creating indirect pressure to introduce big headlining features every so often in order to justify a bump to avoid a renewed perception of stagnation. In the same vein, as already experienced before, with this type of version bump, there will be strong temptation to including breaking changes within its scope, every single time a new one happens – not just because of the seeming opportunity to clean house, but partly also because a perceived need to break certain things more easily makes the case for bumping the version.
+
+Moving to version 42 combines (hopefully the best) aspects of all of these options. It retains a lot of the technical non-cost of option 1, by being essentially indistinguishable from option 1 at the XS level. Outwardly it is closest to option 2, in that the version changes without being motivated by a particular headlining feature, but unlike option 2, it does address the optics of changing the version. It just does so in a very different way from option 3: rather than justifying the bump by making big changes to the language, it highlights the fact that important changes have already been happening and the language has already been evolving, all along.
+
+## Open Issues
+
+1. The environment variable superseding leaves us with an awkward priority cascade for `PERL_LIB` > `PERL5LIB` > `PERLLIB`. However, the last one of these is already mainly historical.
+
+   Maybe we want to reuse `PERLLIB` by flipping the priorities instead of introducing `PERL_LIB`. In that case maybe the counterpart to `PERL5OPT` should be named `PERLOPT`, not `PERL_OPT`, given that these two variables are somewhat complementary. But this would miss the opportunity to tidy things up into “everything is named `PERL_*` (except `PERLIO`)”.
+
+   Maybe we want to just remove `PERLLIB`.
+
+   Maybe we just shrug and accept the awkwardness.
+
+1. Something needs to be done about version.pm:
```

Comment:
Corollary: '42.1' parses as 42.100.0, because you are no longer using two dots. This might be a problem, and we cannot change how these numbers get parsed as versions.

Reply:
Yeah, that’s not changing. As acknowledged elsewhere, the next point release will (in `$]` parlance) be 42.001.

Reply:
I was more stating that (as this section made clearer) we might run into additional confusion if we start trying to popularize 2-segment version numbers which will mostly break if we forget the v.